src/libs/auth.ts

```diff
@@ -31,6 +32,18 @@ import {
 import * as Utils from 'src/libs/utils';
 import { v4 as uuid } from 'uuid';
 
+export interface OidcUser extends User {
+  profile: B2cIdTokenClaims;
+}
+
+export interface B2cIdTokenClaims extends IdTokenClaims {
+  email_verified?: boolean | undefined;
```

Comment:
any reasion who these props are both maked as optional?: type AND unioned with type | undefined?  Doesn't optional?: type already do type | undefined for you?

Reply:
I think the thought process is that we should be marking it as optional if the value could be uninitialized. If it is a union type with undefined, that means that it is either deliberately set as undefined either during initialization or at another time. Does that seem accurate @nawatts ?

Following that logic, since these come back from b2c and should be effectively read only, these should only be marked as optional and not be a union type.

Reply:
Strictly speaking, the `?` suggests that the field may not be present at all in the object.

With our current TypeScript configuration (where [exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) is false), TS accepts that type as matching cases where the field is present but set to `undefined`.

So, practically, these are equivalent and the latter is more frequently used in other Terra UI code.

```ts
field?: Type | undefined
```

```ts
field?: Type
```

However, if the field _is_ always present, but may be set to `undefined`, then I would suggest
```ts
field: Type | undefined
```